src/widgets/allergies/allergy-form.component.tsx

```diff
@@ -13,183 +13,169 @@ import {
 import { createErrorHandler } from "@openmrs/esm-error-handling";
 import { useCurrentPatient } from "@openmrs/esm-api";
 import dayjs from "dayjs";
+import { capitalize } from "lodash-es";
 import { DataCaptureComponentProps } from "../shared-utils";
-
-const DRUG_ALLERGEN_CONCEPT: string = "162552AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA";
-const ENVIRONMENTAL_ALLERGEN_CONCEPT: string =
-  "162554AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA";
-const FOOD_ALLERGEN_CONCEPT: string = "162553AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA";
-const MILD_REACTION_SEVERITY_CONCEPT: string =
-  "1498AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA";
-const MODERATE_REACTION_SEVERITY_CONCEPT: string =
-  "1499AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA";
-const SEVERE_REACTION_SEVERITY_CONCEPT: string =
-  "1500AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA";
+import { AllergyData, AllergicReaction, Allergen } from "../types";
 
 export default function AllergyForm(props: AllergyFormProps) {
-  const [codedAllergenUuid, setCodedAllergenUuid] = React.useState(null);
-  const [allergenType, setAllergenType] = React.useState(null);
-  const [allergensArray, setAllergensArray] = React.useState(null);
-  const [allergyReaction, setAllergyReaction] = React.useState(null);
-  const [enableCreateButtons, setEnableCreateButtons] = React.useState(true);
-  const [enableEditButtons, setEnableEditButtons] = React.useState(true);
-  const [patientAlleryReaction, setPatientAllergyReaction] = React.useState([]);
-  const [comment, setComment] = React.useState("");
-  const [selectedAllergyCategory, setSelectedAllergyCategory] = React.useState(
+  const formRef = useRef<HTMLFormElement>(null);
+  const [viewEditForm, setViewEditForm] = useState(false);
```

Comment:
`view` is a verb, which makes it sound like a function name. Since it's a boolean, it should have a yes/no question for a name, like `isEditFormActive`.